src/gui/accountstate.h

```diff
@@ -199,6 +204,7 @@ protected Q_SLOTS:
     QPointer<ConnectionValidator> _connectionValidator;
     QByteArray _notificationsEtagResponseHeader;
     QByteArray _navigationAppsEtagResponseHeader;
+    QSharedPointer<PushNotifications> _pushNotifications = nullptr;
```

Comment:
Yes, use a raw pointer here. And pass the account state as parent to the object at construction time.